likeminds-feed-android-core/src/main/java/com/likeminds/feed/android/core/post/detail/view/LMFeedPostDetailFragment.kt

```diff
@@ -1946,6 +1984,317 @@ open class LMFeedPostDetailFragment :
+                    LMFeedViewUtils.showShortSnack(
+                        binding.root,
+                        resources.getQuantityString(
+                            R.plurals.lm_feed_please_select_at_most_d_options,
+                            multipleSelectNumber,
+                            multipleSelectNumber
+                        )
+                    )
+                    return
+                }
+            }
+
+            PollMultiSelectState.AT_LEAST -> {
+                if (selectedOptionsCount < multipleSelectNumber) {
+                    LMFeedViewUtils.showShortSnack(
+                        binding.root,
+                        resources.getQuantityString(
+                            R.plurals.lm_feed_please_select_at_least_d_options,
+                            multipleSelectNumber,
+                            multipleSelectNumber
+                        )
+                    )
+                    return
+                }
+            }
+        }
+        onValidationSuccess()
+    }
+
+    //callback when the poll edit vote button is clicked
+    override fun onPostEditPollVoteClicked(position: Int, postViewData: LMFeedPostViewData) {
+        super.onPostEditPollVoteClicked(position, postViewData)
+
+        binding.rvPostDetails.apply {
+            val attachment = postViewData.mediaViewData.attachments.firstOrNull() ?: return
+            val pollViewData = attachment.attachmentMeta.poll ?: return
+
+            //update the poll view data
+            val updatedPollViewData = pollViewData.toBuilder()
+                .isPollSubmitted(false)
+                .build()
+
+            //update the attachments with the updated poll view data
+            val updatedAttachments = listOf(
+                attachment.toBuilder()
+                    .attachmentMeta(
+                        attachment.attachmentMeta.toBuilder()
+                            .poll(updatedPollViewData)
+                            .build()
+                    )
+                    .build()
+            )
+
+            //update the post view data
+            val updatedPostViewData = postViewData.toBuilder()
+                .mediaViewData(
+                    postViewData.mediaViewData.toBuilder()
+                        .attachments(updatedAttachments)
+                        .build()
+                )
+                .build()
+
+            updateItem(postDataPosition, updatedPostViewData)
+        }
+    }
+
+    //callback when the poll option is clicked
+    override fun onPollOptionClicked(
+        pollPosition: Int,
+        pollOptionPosition: Int,
+        pollOptionViewData: LMFeedPollOptionViewData
+    ) {
+        super.onPollOptionClicked(
+            pollPosition,
+            pollOptionPosition,
+            pollOptionViewData
+        )
+
+        val postViewData = binding.rvPostDetails.getItem(pollPosition) as LMFeedPostViewData
```

Comment:
Can we move this whole function in some util, so that we can use same code in feed and here.